dhall/src/Dhall/Binary.hs

```diff
@@ -335,8 +335,8 @@ decodeExpressionInternal decodeEmbed = go
                                     _ -> do
                                         Decoding.decodeNull
 
-                                        xs <- replicateDecoder (len - 2) go
-                                        return (ListLit Nothing (Data.Sequence.fromList xs))
+                                        xs <- Data.Sequence.replicateA (len - 2) go
+                                        return (ListLit Nothing xs)
```

Comment:
Did you test whether this is beneficial? I had tried this before, but it seemed to actually slow things down: https://github.com/dhall-lang/dhall-haskell/issues/1804#issuecomment-632918489

Reply:
Thanks for your response in https://github.com/dhall-lang/dhall-haskell/issues/1804#issuecomment-644338163 @bgamari!

It's well possible that when I benchmarked this before, I was mislead by my noisy machine! I any case, the effect will not be huge.